### PR TITLE
Storybook

### PR DESCRIPTION
**Pull request by duggiemitchell**
## Cute animal pic
_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._
photo-1538318514451-db4272ee0fc8

## Link to Trello card (if applicable)
https://trello.com/c/apFqpTbJ

## Description
Adding storybook for component documentation.

## Steps to test/reproduce
Run storybook using `./derrick storybook`
Further reading on creating a story: 

[Naming components and hierarchy](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy)
[How to document components](https://storybook.js.org/docs/react/writing-docs/introduction)

### CODE REVIEW

File: derrick [hunk 1]
```diff
@@ -48,6 +48,9 @@ case "$1" in
     "test:js")
         dcrq yarn test:js "${@:2}"
         ;;
+    "storybook")
+        dcrq yarn run storybook "${@:2}"
+        ;;
```
Comment (duggiemitchell) [inline derrick:53]: @jgerigmeyer Can you point out what's missing for this work. The idea is to be able to run storybook from the container, but I feel like there may be another set needed here.    
`$ ./derrick storybook
yarn run v1.22.5
$ start-storybook -p 6006 
/bin/sh: 1: start-storybook: not found
`

Comment (jgerigmeyer) [reply derrick]: @duggiemitchell Fixed!

File: .storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import 'sass/app.scss';
+
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
+};
+
+export const decorators = [
+  (Story) => (
+    <MemoryRouter>
+      <IconSettings
```
Comment (duggiemitchell) [inline .storybook/preview.js:19]: eek, looks like docs for [Icon Settings](https://react.lightningdesignsystem.com/components/icon-settings/) aren't working but this makes sense!

Comment (jgerigmeyer) [reply .storybook/preview.js:19]: @duggiemitchell Huh, good call -- weird about the docs.

I basically tried to re-create the "wrapping" setup that we're using in `src/js/index.tsx` (https://github.com/oddbird/Metecho/blob/main/src/js/index.tsx#L179-L191) or test files. One thing I didn't worry about is whether this setup will break for a component that tries to access the Redux store. If we want to document such a component, we'll have to test it out and see what the best workaround is (probably a similar approach as our test files, mocking out a minimal Redux store for the specific story).

**Review by jgerigmeyer — APPROVED** [review]
LGTM! I'll let @duggiemitchell or @stacyk or @dvdherron merge if it's 👍

File: src/stories/fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  assigned_dev: sampleGitHubUser1,
+  assigned_qa: null,
+  status: TASK_STATUSES.IN_PROGRESS,
+  currently_submitting_review: false,
+  review_submitted_at: '2019-03-01T19:47:49Z',
+  review_valid: true,
+  review_status: REVIEW_STATUSES.APPROVED,
+  review_sha: '617a512',
+  org_config_name: 'dev',
+};
```
Comment (jgerigmeyer) [inline src/stories/fixtures.ts:172]: @duggiemitchell @stacyk @dvdherron If you three prefer to split this up into separate files (e.g. `/fixtures/epics.ts`, `/fixtures/tasks.ts` etc) that's fine with me -- your call.

**Review by dvdherron — APPROVED** [review]
LGTM! Thanks for the fixtures help @jgerigmeyer. That's going to save a lot of time.

File: src/stories/pages/404.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import { FourOhFour } from './404.stories';
+
+<Meta title="Pages/404/Intro" component={FourOhFour} />
+
+# 404
+
+A page view also known as an “error page” or “Page Not Found” page. This page
+indicates that the user reached the domain they requested, but the URL path
+provided no information.
```
Comment (duggiemitchell) [inline src/stories/pages/404.stories.mdx:12]: @dvdherron  Not a blocker for this pr, but I'm open to any ideas you have on component documentation. This is just something I copy/pasta'd from the web and we can include those changes in the [404 card](https://trello.com/c/f0jSvig0/93-metecho-404) in another branch,

File: .storybook/decorators.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+export const withRedux = (state = {}) => (Story: React.ComponentClass) => {
+  const store = configureStore([thunk, storyMiddleware])(state);
+  return (
+    <Provider store={store}>
+      <Story />
+    </Provider>
+  );
+};
```
Comment (duggiemitchell) [inline .storybook/decorators.tsx]: @jgerigmeyer  I think I get why we need to wrap some components with this decorator (ie the TaskTable that gets data from the store). Wondering if when this is added, we don't need the IconSetting wrapper?

File: src/stories/pages/404.stories.tsx [hunk 6]
```diff
@@ -5,14 +5,14 @@ import FourOhFourComponent from '~js/components/404';
...
 };
 
 const Template: Story<ComponentProps<typeof FourOhFourComponent>> = (args) => (
   <FourOhFourComponent {...args} />
 );
 
 export const FourOhFour = Template.bind({});
-
 FourOhFour.args = {
   message: 'Lorem Ipsum',
```
Comment (duggiemitchell) [inline src/stories/pages/404.stories.tsx:17]: @jgerigmeyer not a big deal for this pr, but I'm wondering if there's another decorator missing to account for components that rely on having the `routes` available like this 404 if we called this without a `message` arg.

File: src/stories/pages/TasksTable.stories.tsx [hunk 7]
```diff
@@ -24,9 +35,9 @@ TasksTable.args = {
       description_rendered: '<p>This is <em>safely</em> rendered Markdown.</p>',
       epic: '3Lw7OwK',
       assigned_dev: {
-        id: '12345',
-        login: 'username',
-        avatar_url: 'https://avatars0.githubusercontent.com/u/someId?v=4',
+        id: '123456',
+        login: 'TestGitHubUser',
+        avatar_url: 'https://randomuser.me/api/portraits/men/1.jpg',
       },
```
Comment (duggiemitchell) [inline src/stories/pages/TasksTable.stories.tsx]: 🤵‍♂️

**Review by duggiemitchell — COMMENTED** [review]
(though I cannot approve) LGTM left some questions/comments 👍🏾

File: src/stories/pages/EpicsTable.stories.tsx [hunk 8]
```diff
@@ -13,33 +15,32 @@ const Template: Story<ComponentProps<typeof EpicsTableComponent>> = (args) => (
...
+        'https://github.com/test/test-repo/compare/main...feature/my-epic',
       pr_url: null,
       pr_is_open: false,
       pr_is_merged: false,
       has_unmerged_commits: false,
       currently_creating_pr: false,
       currently_fetching_org_config_names: false,
       github_users: [],
-      status: 'Planned',
+      status: EPIC_STATUSES.PLANNED,
```
[the remark this answers is not in the document]
Comment (jgerigmeyer) [reply src/stories/pages/EpicsTable.stories.tsx]: No, you'll see in `constants.ts` (https://github.com/oddbird/Metecho/blob/main/src/js/utils/constants.ts#L58) that `EPIC_STATUSES.PLANNED` === `'Planned'`. This is a best-practice attempt to not have "magic" strings strewn throughout the code, but to consistently use one constant. That way if the Python code ever changes the value of the "Planned" status we would only have to make the change in one place in the JS. Also, it dramatically cuts down on the chance that a small typo in one string could break things.

File: src/stories/pages/EpicsTable.stories.tsx [hunk 9]
```diff
@@ -49,21 +50,38 @@ EpicsTable.args = {
...
       branch_diff_url:
-        'https://github.com/test/test-repo/compare/my-project/my-project-my-epic',
-      pr_url: null,
-      pr_is_open: false,
+        'https://github.com/test/test-repo/compare/main...feature/midyear-project-saturn',
+      pr_url: 'https://github.com/test/test-repo/pull/1234',
+      pr_is_open: true,
       pr_is_merged: false,
-      has_unmerged_commits: false,
+      has_unmerged_commits: true,
```
[the remark this answers is not in the document]
Comment (jgerigmeyer) [reply src/stories/pages/EpicsTable.stories.tsx]: Hm, I'm not sure what you mean by "default". None of these are really "optional" props -- it's a list of Epic objects, and an epic can either have `has_unmerged_commits: false` or `has_unmerged_commits: true`. I guess `false` is kind of the "default" state in a sense. I think we're showing the list of epics in a couple different states -- if you'd prefer to break that out into a base story with the initial state of an epic that's just been created, then different stories for epics in different states -- that's fine with me! 👍

File: .storybook/decorators.tsx (hunk 5 above)
Comment (jgerigmeyer) [reply .storybook/decorators.tsx]: The decorators returned from `preview.js` are global decorators (https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) that are always used for all stories. This file is for custom decorators that we can use or not use as we like -- but even when used, they are _in addition to_ the global decorators, not replacing.

(`TaskTable` doesn't actually use the store, but it does have the `AssignUsersModal` embedded, and that child component accesses the `user` object from the store.)

File: src/stories/pages/404.stories.tsx (hunk 6 above)
Comment (jgerigmeyer) [reply src/stories/pages/404.stories.tsx:17]: In `preview.js`, we're using the `<MemoryRouter>` in the global decorator, which allows using react-router things (e.g. `<Link>`) in our stories -- so this works as-is.

**Pull request merged**